### PR TITLE
Add TF_ORGANIZATION env var support

### PR DESCRIPTION
**Pull request by sebasslash**
`TF_ORGANIZATION` will serve as a fallback for configuring an organization in the `cloud`
block. This is the first step to make it easier for users wanting to configure Terraform
programmatically.

## Testing this feature
1. Pull this branch locally and run `go install .` which will install the binary in `$GOPATH/bin/terraform` 
1. Export `TF_ORGANIZATION` so Terraform can consume the var: 
```sh
export TF_ORGANIZATION="my-awesome-org"
```
3. Create a simple Terraform configuration, omitting `organization` from the `cloud` block. We'll use the `tfe` provider in this example. 
```hcl
terraform {
  cloud {
    hostname = "my-tfe.host.com"
    workspaces {
      name = "test"
    }
  }
}

provider "tfe" {
  hostname = "my-tfe.host.com"
  token = "my-tfe-token"
}

resource "tfe_workspace" "test" {
  name = "my-awesome-workspace"
  organization = "my-awesome-organization"
}
```
4. Run `$GOPATH/bin/terraform init` and then `$GOPATH/bin/terraform apply` ! 

### What if both values are set?

An organization specified in the cloud block should always take precedence over `TF_ORGANIZATION`. Try setting an organization in your configuration while `TF_ORGANIZATION` is set in your environment to test this. 

### Run Unit Tests
```sh
go test ./internal/cloud/ -run TestCloud_PrepareConfigWithEnvVars
go test ./internal/cloud/ -run TestCloud_configWithEnvVars
```

### CODE REVIEW

**Review by barrettclark — CHANGES_REQUESTED** [review]
Thanks for working on this! ✨ I pulled it down and ran through some test cases, and I think I found something 🤔 

When I test this with no organization set in the cloud config block and `TF_ORGANIZATION` set in the environment, I get this message:

```
Initializing Terraform Cloud...
organization
  The name of the organization containing the targeted workspace(s).
```

Also, the tests look to be using a config that should error as invalid. It looks like both a name _and_ tags are specified in the workspaces section of the config. Am I mis-reading that? Totally possible -- go tests are still hard for me to read. 😬 

Finally, it would be great to see a test that sets an environment variable and does not set an organization in the config.

File: internal/cloud/backend_test.go [hunk 1]
```diff
@@ -146,6 +146,124 @@ func TestCloud_PrepareConfig(t *testing.T) {
...
+		for k, v := range tc.vars {
+			switch k {
+			case "TF_ORGANIZATION":
+				val := tc.config.GetAttr("organization")
+				if !val.IsNull() && val.AsString() != "" && val.AsString() != b.organization {
+					t.Fatalf("%s: organization not valid: %s, expected: %s", name, b.organization, val.AsString())
+				} else if val.IsNull() && b.organization != v {
+					t.Fatalf("%s: organization not valid: %s, expected: %s", name, b.organization, v)
+				}
+			}
```
Comment (alisdair) [inline internal/cloud/backend_test.go:262]: I found the logic in this test hard to follow, as it's just about as complex as the logic it's trying to cover. I wonder if we could alter the table used to derive the tests in order to make this simpler.

A common pattern in Terraform tests is a `wantFoo` field in the table struct, defining an expected output value. For these tests, maybe a `wantOrganization` could specify what we expect the code to do with the given inputs, so that the test code is just checking that `b.organization == tc.wantOrganization`. Does that make sense to you?

**Review by alisdair — COMMENTED** [review]
The code looks good to me! 👍 I have a suggestion inline about tests. It would also be good to update the associated documentation in the same commit: https://github.com/hashicorp/terraform/blob/main/website/docs/cli/cloud/settings.mdx

File: internal/cloud/backend.go [hunk 2]
```diff
@@ -152,8 +152,13 @@ func (b *Cloud) PrepareConfig(obj cty.Value) (cty.Value, tfdiags.Diagnostics) {
 		return obj, diags
 	}
 
+	// check if organization is specified in the config.
 	if val := obj.GetAttr("organization"); val.IsNull() || val.AsString() == "" {
-		diags = diags.Append(invalidOrganizationConfigMissingValue)
+		// organization is specified in the config but is invalid, so
+		// we'll fallback on TF_ORGANIZATION
+		if val := os.Getenv("TF_ORGANIZATION"); val == "" {
+			diags = diags.Append(invalidOrganizationConfigMissingValue)
```
Comment (alisdair) [inline internal/cloud/backend.go:160]: This error contains an unevaluated format string, resulting in the following incorrect output:

init

**Review by alisdair — APPROVED** [review]
Looks good to me! When testing locally I spotted a bug in the error diagnostic displayed if the organization is unset, which it would be nice to fix before merge.

**Review by barrettclark — APPROVED** [review]
